cookbooks/equilibrium_grain_size/plugins/equilibrium_grain_size.cc

```diff
@@ -0,0 +1,1750 @@
+          temperature_scaling_index = dT_vs_depth_profile.get_column_index_from_name("temperature_scaling");
+        }
+
+      // Get column for crustal depths
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"));
+      crustal_boundary_depth.initialize(surface_boundary_set, 1);
+
+      this->get_signals().post_stokes_solver.connect([&](const SimulatorAccess<dim> &,
+                                                         const unsigned int ,
+                                                         const unsigned int ,
+                                                         const SolverControl &,
+                                                         const SolverControl &)
+      {
+        this->update();
+      });
+
+      this->get_signals().post_advection_solver.connect([&](const SimulatorAccess<dim> &,
+                                                            const unsigned int ,
+                                                            const unsigned int ,
+                                                            const SolverControl &)
+      {
+        this->update();
+      });
+
+      n_material_data = material_file_names.size();
+      for (unsigned i = 0; i < n_material_data; i++)
+        {
+          if (material_file_format == perplex)
+            material_lookup.push_back(std::shared_ptr<MaterialUtilities::Lookup::MaterialLookup>
+                                      (new MaterialUtilities::Lookup::PerplexReader(datadirectory+material_file_names[i],
+                                                                                    /*use_bilinear_interpolation*/ true,
+                                                                                    this->get_mpi_communicator())));
+          else if (material_file_format == hefesto)
+            material_lookup.push_back(std::shared_ptr<MaterialUtilities::Lookup::MaterialLookup>
+                                      (new MaterialUtilities::Lookup::HeFESToReader(datadirectory+material_file_names[i],
+                                                                                    datadirectory+derivatives_file_names[i],
+                                                                                    /*use_bilinear_interpolation*/ true,
+                                                                                    this->get_mpi_communicator())));
+          else
+            AssertThrow (false, ExcNotImplemented());
+        }
+    }
+
+
+
+    template <int dim>
+    void
+    EquilibriumGrainSize<dim>::update()
+    {
+      if (use_depth_dependent_viscosity)
+        {
+          std::vector<std::unique_ptr<internal::FunctorBase<dim>>> lateral_averaging_properties;
+          lateral_averaging_properties.push_back(std::make_unique<internal::FunctorDepthAverageUnscaledViscosity<dim>>());
```

Comment:
Or maybe even easier, just create a one-element array:
```suggestion
          const std::vector<std::unique_ptr<internal::FunctorBase<dim>>> lateral_averaging_properties
          = { std::make_unique<internal::FunctorDepthAverageUnscaledViscosity<dim>>() };
```